src/server/apis/dash-data.ts

```diff
@@ -231,11 +231,67 @@ export class DashData {
     return {
       // Sort newest first.
       outgoingPrs: outgoingPrs.sort((a, b) => b.createdAt - a.createdAt),
-      incomingPrs,
+      incomingPrs: sortIncomingPRs(incomingPrs),
     };
   }
 }
 
+/**
+ * Sorts incoming PRs into how they should be displayed.
+ */
+function sortIncomingPRs(prs: api.PullRequest[]): api.PullRequest[] {
+  function getLatestEventTime(pr: api.PullRequest): number {
+    let latest = pr.createdAt;
+    for (const event of pr.events) {
+      switch (event.type) {
+        case 'NewCommitsEvent':
+          if (event.lastPushedAt > latest) {
+            latest = event.lastPushedAt;
+          }
+          break;
+        case 'MentionedEvent':
+          if (event.mentionedAt > latest) {
+            latest = event.mentionedAt;
+          }
+          break;
+        case 'MyReviewEvent':
+          // Ignore my review events. Shouldn't impact ordering since it's an
+          // event generated by you.
+          break;
+        default:
+          break;
+      }
+    }
+    return latest;
+  }
+
+  // Sort descending by latest event time.
+  const timeSorted =
+      prs.sort((a, b) => getLatestEventTime(b) - getLatestEventTime(a));
+  const result = [];
+
+  const notActionable = [
```

Comment:
Out of the scope for this PR, but does it make sense to add an `actionable` field to this data? I just worry that any changes to this list will need changes in a few different places.

Reply:
Yeah maybe. Originally I wanted to keep it out of the server, but now its in the server.